cmd/src/users_verify.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"strings"
+)
+
+const SubCommand = "verify-emails"
+
+type VerifyEmailJson struct {
+	ID    string `json:"ID"`
+	Email string `json:"Email"`
+}
+
+func init() {
+	usage := `
+Examples:
+
+  Verify one or more user account(s):
+
+    	$ src users ` + SubCommand + ` -verify-json='[{"ID": "ID1", "Email": "email1"},...]'
+
+`
+
+	flagSet := flag.NewFlagSet(SubCommand, flag.ExitOnError)
+	usageFunc := func() {
+		fmt.Fprintf(flag.CommandLine.Output(), "Usage of 'src users %s':\n", flagSet.Name())
+		flagSet.PrintDefaults()
+		fmt.Println(usage)
+	}
+	var (
+		verifyJsonFlag = flagSet.String("verify-json", "",
+			`The JSON array of id-email descriptors of those user/emails that should be marked as verified. (required)`)
+		apiFlags = newAPIFlags(flagSet)
+	)
+
+	handler := func(args []string) error {
+		flagSet.Parse(args)
+
+		dec := json.NewDecoder(bytes.NewBufferString(*verifyJsonFlag))
+
+		var verifyList []VerifyEmailJson
+		if err := dec.Decode(&verifyList); err != nil {
+			return err
+		}
+
+		vars := map[string]interface{}{}
+
+		// translate the provided index of `verifyList` into variables for its `id` and `email` payload
+		indexToKeys := func(idx int) (string, string) {
+			return fmt.Sprintf("id%09d", idx), fmt.Sprintf("email%09d", idx)
+		}
+
+		for idx, verifyObj := range verifyList {
+			idKey, emailKey := indexToKeys(idx)
+			vars[idKey] = verifyObj.ID
+			vars[emailKey] = verifyObj.Email
+		}
+
+		query := `mutation VerifyUserEmails(`
+		for k := range vars {
+			varType := "String!"
+			if strings.HasPrefix(k, "id") {
+				varType = "ID!"
+			}
+			query += fmt.Sprintf("$%s: %s,", k, varType)
```

Comment:
I observed that [`strings.Builder`](https://godoc.org/strings#Builder) is a thing, but it wasn't clear if the code noise is worth the reduction in memory copying; I welcome input about that

Reply:
Meh, only if it makes the code more readable or performance is sensitive. I haven't used it before personally. I think my previous comment about text/template is more important and would probably make this question go away.